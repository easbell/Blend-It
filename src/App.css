.App {
  text-align: center;
  background-color: #82204A;
}

h1 {
  font-family: Sacramento;
  font-size: 76px;
  padding: 10px;
  color: #E8DB7D;
}

h2 {
  color: #E8DB7D;
  
}