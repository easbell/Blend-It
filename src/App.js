import React, { Component } from 'react';
import SmoothieContainer from './SmoothieContainer.js'
import {IngredientContainer} from './IngredientContainer.js'
import ChosenIngredientList from './ChosenIngredientList.js'

class App extends Component {
  constructor() {
    super();

    this.state = {
      smoothies: [],
      ingredients: {},
      error: '',
      chosenIngredients: [],
      filteredSmoothies: []
    }
  }
  
  componentDidMount() {
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/smoothies')
    .then(response => response.json())
    .then(result => {
      this.setState({
        smoothies: result.smoothies,
        filteredSmoothies: result.smoothies
      })
    })
    .catch(error => {
      this.setState({error: error.message})
    })
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/ingredients')
      .then(response => response.json())
      .then(data => {
        this.setState({
          ingredients: data.ingredients
        })
      })
      .catch(error => {
      this.setState({error: error.message})
      })
  }

  chooseIngredients = (ingredient) => {
    const updatedIngredients = [ ...this.state.chosenIngredients, ingredient]
    this.setState({ chosenIngredients: updatedIngredients }, () => {
      this.filterSmoothies();
    })
  }

  removeIngredient = (ingredient) => {
    const updatedIngredients = this.state.chosenIngredients.filter(chosenIngredient => {
      return chosenIngredient !== ingredient
    })
    this.setState({ chosenIngredients: updatedIngredients, filteredSmoothies: this.state.smoothies }, () => {
      if(this.state.chosenIngredients.length) {
        this.filterSmoothies();
      }
    })
  }

  filterSmoothies = () => {
    const { chosenIngredients } = this.state;
    const matching = this.state.filteredSmoothies.filter(smoothie => {
      let matched = false;
       chosenIngredients.forEach(chosen=> {
          matched = smoothie.ingredients.includes(chosen);
       })
       return matched;
    })
    this.setState({ filteredSmoothies: matching });
  }

  render() {
    return (
      <div className="App">
        {this.state.error && <p>{this.state.error}</p>}
        <h1>Blend It!</h1>
        <h2>Choose Your Ingredients:</h2>
        {Object.keys(this.state.ingredients).length > 0 && 
          <IngredientContainer 
          ingredients={this.state.ingredients} 
          chooseIngredients={this.chooseIngredients}
          removeIngredient={this.removeIngredient}
          />
        }
        {this.state.chosenIngredients.length > 0 &&
          <ChosenIngredientList ingredients={this.state.chosenIngredients} />
        }
        <SmoothieContainer 
          smoothies={this.state.filteredSmoothies}
          chosenIngredients={this.state.chosenIngredients}
        />
      </div>
    )
  }
}

export default App;
