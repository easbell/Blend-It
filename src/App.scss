.App {
  text-align: center;
  background-color: #82204A;
}

h1 {
  font-family: Sacramento;
  font-size: 76px;
  padding: 10px;
  margin-bottom: 5px;
  color: #E8DB7D;
}

h2 {
  color: #558C8C;
  letter-spacing: 2px;
  font-size: 32px;
  margin: 0;
}