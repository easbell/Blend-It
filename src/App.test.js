import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

const mockSmoothies = [{
id: 1,
name: "Babe Ruth Strawberry Pineapple Banana Smoothie",
ingredients: [
"strawberries",
"pineapple",
"banana",
"orange juice",
"yogurt",
"spinach",
"chia seeds",
"flaxseed"
],
recipe: [
"1 cup strawberries",
"1/2 cup pineapple",
"1 banana",
"2 cups orange juice",
"1/2 cup greek yogurt",
"1 cup spinach, optional",
"1 tablespoon chia or flaxseed, optional",
"ice"
],
servings: 4,
source: "https://www.modernhoney.com/6-healthy-superfood-smoothies/",
img: "images/babe-ruth.jpg"
},
{
id: 2,
name: "Cherry Almond Smoothie",
ingredients: [
"milk",
"cherries",
"raw almonds",
"honey",
"cinnamon",
"almond extract",
"coconut oil"
],
recipe: [
"1 cup almond milk",
"1 cup cherries",
"3 tablespoons raw almonds",
"2 tablespoons raw honey",
"1 teaspoon cinnamon",
"1/2 teaspoon almond extract",
"1 tablespoon coconut oil"
],
servings: 1,
source: "https://bakedbree.com/cherry-almond-smoothies",
img: "images/cherry-almond.jpg"
}];

const mockIngredients = {
  fruit: [
  "banana",
  "pineapple",
  "mango"
  ],
  vegetables: [
  "spinach",
  "kale",
  "carrot"
  ],
  bases: [
  "milk",
  "yogurt",
  "coconut water"
  ],
  extras: [
  "chia seeds",
  "ginger",
  "cinnamon"
  ]
}

describe('App', () => {
  const wrapper = shallow(
    <App />
  );

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper default states', () => {
    expect(wrapper.state()).toEqual({smoothies: [], ingredients: {}, error: '', chosenIngredients: [], filteredSmoothies: []})
  })

  it('should update the chosen ingredients when ingredient is selected', () => {
    expect(wrapper.state("chosenIngredients")).toEqual([])
    wrapper.instance().chooseIngredients("milk")
    expect(wrapper.state("chosenIngredients")).toEqual(["milk"])
  });

  it('should update the chosen ingredients when ingredient is de-selected', () => {
    expect(wrapper.state("chosenIngredients")).toEqual(["milk"])
    wrapper.instance().removeIngredient("milk")
    expect(wrapper.state("chosenIngredients")).toEqual([])
  });

  // it('should only show smoothies which match each chosen ingredient', () => {
  //   expect(wrapper.state("chosenIngredients")).toEqual(["oats"])
  //   wrapper.instance().filterSmoothies()
  //   expect(wrapper.state("smoothies").length).toEqual(3)
  // });
});


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
