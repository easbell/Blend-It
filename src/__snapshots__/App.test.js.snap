// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot with all data passed in 1`] = `
<div
  className="App"
>
  <h1>
    Blend It!
  </h1>
  <h2>
    Choose Your Ingredients
  </h2>
  <ChosenIngredientList
    ingredients={Array []}
  />
  <SmoothieContainer
    smoothies={Array []}
  />
</div>
`;
